### PR TITLE
Add reset user data functional , getById and having data from the beg…

### PR DESCRIPTION
**Pull request by ArthurKhachatryanRepositories**
…ining

### CODE REVIEW

File: src/main/java/com/epam/team1/emotional/help/controller/SignupController.java [hunk 1]
```diff
@@ -26,16 +26,13 @@ public class SignupController {
...
+        return signupService.saveUser(signupUserRequestDTO);
     }
 
     @GetMapping("/confirm-email/{code}")
-    public MessageResponse confirmEmail(@PathVariable("code") String code) {
-        MessageResponse messageResponse = signupService.confirmEmail(code);
-        log.error("email is conformed with code " + code);
-        return messageResponse;
+    public SignupConformedResponseDto confirmEmail(@PathVariable("code") String code) {
+        return signupService.confirmEmail(code);
```
Comment (jndr29384) [inline src/main/java/com/epam/team1/emotional/help/controller/SignupController.java:36]: Please, add log here, and probably a good idea to check that code is not empty.

File: src/main/java/com/epam/team1/emotional/help/service/SignupService.java [hunk 2]
```diff
@@ -38,41 +46,47 @@ public Optional<User> findByEmail(String email) {
...
+        String confirmLink = confirmLinkIP + "confirm/" + code;
+        String mailText = "Click on this link to confirm your email \n" + confirmLink;
+        mailService.send(email, subject, mailText);
     }
 
     private AccessCode generateAccessCode(User user, String code) {
         AccessCode accessCode = user.getAccessCode();
         if (accessCode != null) {
-            AccessCode byId = accessCodeRepository.findById(accessCode.getId()).orElseThrow(()->new NotFoundException("Code does not exist"));
+            AccessCode byId = accessCodeRepository.findById(accessCode.getId()).orElseThrow(() -> new NotFoundException("Code does not exist"));
```
Comment (jndr29384) [inline src/main/java/com/epam/team1/emotional/help/service/SignupService.java]: Rename to `accessCode` or something else

File: src/main/java/com/epam/team1/emotional/help/service/UserService.java [hunk 3]
```diff
@@ -1,11 +1,71 @@
...
+        User currentUser;
+        try {
+            currentUser = ((UserDetailsImplementation) SecurityContextHolder.
+                    getContext().
+                    getAuthentication().
+                    getPrincipal()).getUser();
+        } catch (NullPointerException e) {
+            return null;
+        }
+        return currentUser;
```
Comment (jndr29384) [inline src/main/java/com/epam/team1/emotional/help/service/UserService.java]: Looks really strange

**Review by jndr29384 — CHANGES_REQUESTED** [review]

File: src/main/java/com/epam/team1/emotional/help/service/UserService.java (hunk 3 above)
Comment (ArthurKhachatryanRepositories) [reply src/main/java/com/epam/team1/emotional/help/service/UserService.java]: can I optimize this code in the next branch

**Pull request merged**